lib/Serialization/Serialization.cpp

```diff
@@ -4080,7 +4080,15 @@ class Serializer::DeclSerializer : public DeclVisitor<DeclSerializer> {
   /// \returns the number of entries added.
   size_t addInherited(InheritedTypes inheritedEntries,
                       SmallVectorImpl<uint64_t> &result) {
-    for (const auto &inherited : inheritedEntries.getEntries()) {
+    for (size_t i = 0; i < inheritedEntries.size(); ++i) {
+      // Ensure that we run the `InheritedTypeRequest` before getting the
+      // inherited type. We serialize the inherited type into the module
+      // because we serialize whether the conformance is suppressed as a
+      // separate bit instead of inside the inherited type itself. We thus need
```

Comment:
This reads a little weird to me, maybe it should be something more like "we serialize the inherited type from `getEntry` rather than `getResolvedType` since the former represents a suppressed conformance as a separate bit distinct from the type, which is how we want to serialize it"

Reply:
Good suggestion 👍🏽